### PR TITLE
Added Defaults to `ipfs daemon`

### PR DESCRIPTION
**Pull request by RichardLitt**
Split from https://github.com/ipfs/go-ipfs/pull/2484.

License: MIT
Signed-off-by: Richard Littauer <richard.littauer@gmail.com>

### CODE REVIEW

File: cmd/ipfs/daemon.go [hunk 1]
```diff
@@ -126,16 +126,16 @@ future version, along with this notice. Please move to setting the HTTP Headers.
...
-		cmds.BoolOption(mountKwd, "Mounts IPFS to the filesystem"),
-		cmds.BoolOption(writableKwd, "Enable writing objects (with POST, PUT and DELETE)"),
-		cmds.StringOption(ipfsMountKwd, "Path to the mountpoint for IPFS (if using --mount)"),
-		cmds.StringOption(ipnsMountKwd, "Path to the mountpoint for IPNS (if using --mount)"),
-		cmds.BoolOption(unrestrictedApiAccessKwd, "Allow API access to unlisted hashes"),
-		cmds.BoolOption(unencryptTransportKwd, "Disable transport encryption (for debugging protocols)"),
-		cmds.BoolOption(enableGCKwd, "Enable automatic periodic repo garbage collection"),
-		cmds.BoolOption(adjustFDLimitKwd, "Check and raise file descriptor limits if needed"),
+		cmds.BoolOption(initOptionKwd, "Initialize IPFS with default settings if not already initialized").Default(false),
+		cmds.StringOption(routingOptionKwd, "Overrides the routing option (dht, supernode)").Default("supernode"),
```
Comment (whyrusleeping) [inline cmd/ipfs/daemon.go]: the default is not supernode, lol

Comment (RichardLitt) [inline cmd/ipfs/daemon.go]: What is it? dht? :P

Comment (Kubuxu) [inline cmd/ipfs/daemon.go]: It is DHT.

Comment (RichardLitt) [inline cmd/ipfs/daemon.go]: What does 'supernode' mean here, then? I haven't run across this term before.

Comment (whyrusleeping) [inline cmd/ipfs/daemon.go]: it was an old bit of code we wrote to be a 'centralized' routing system. We kindof abandoned it. The code might still work, but its not used.

Comment (RichardLitt) [inline cmd/ipfs/daemon.go]: Ok, removed. We should remove that old code, then, too.

**Comment by Kubuxu** [comment]
LGTM

**Pull request merged**